packages/itwinui-react/src/core/Table/TableExpandableRow.tsx

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import * as React from 'react';
+import type { Row } from '../../react-table/react-table.js';
+import { Box, useMergedRefs } from '../../utils/index.js';
+import type { Virtualizer, VirtualItem } from '@tanstack/react-virtual';
+
+export const TableExpandableRow = <T extends Record<string, unknown>>(props: {
+  row: Row<T>;
+  subComponent?: (row: Row<T>) => React.ReactNode;
+  virtualItem?: VirtualItem<Element>;
+  virtualizer?: Virtualizer<Element, Element>;
+  isDisabled: boolean;
+  tableRowRef?: React.Ref<HTMLDivElement>;
+}) => {
+  const {
+    row,
+    subComponent,
+    virtualItem,
+    virtualizer,
+    isDisabled,
+    tableRowRef,
+  } = props;
+  const mergedProps = {
```

Comment:
nit: this extra variable looks unnecessary, the props can be directly set on the `Box` instead

Reply:
I have set this directly on the `Box`.